assets/js/components/PresenterStreamArea.tsx

```diff
@@ -9,35 +9,36 @@ type PresenterStreamAreaProps = {
 };
 
 export const presenterStreams: { [key: string]: MediaStream } = {};
-let isConnected = false;
+let webrtc: any = null;
 
 const PresenterStreamArea = ({ username, presenters, eventChannel }: PresenterStreamAreaProps) => {
   const [streamsAvailable, setStreamsAvailable] = useState<{ [key: string]: boolean }>({});
 
   useEffect(() => {
-    // TODO: self-view
-    // implement changing audio/video source
-    if (!isConnected && presenters.includes(username)) {
-      connectWebrtc(eventChannel, username, streamsAvailable, setStreamsAvailable);
-      isConnected = true;
-    } else if (isConnected && !presenters.includes(username)) {
-      // TODO: leave WebRTC connection
+    if (webrtc == null && presenters.includes(username)) {
+      webrtc = connectWebrtc(eventChannel, username, streamsAvailable, setStreamsAvailable);
+    } else if (webrtc != null && !presenters.includes(username)) {
+      leaveWebrtc(webrtc, username, streamsAvailable, setStreamsAvailable, eventChannel);
+      webrtc = null;
     }
   }, [presenters]);
 
-  return (
-    <div>
+  return presenters.includes(username) ? (
+    <>
       {presenters.map((presenter) => {
         return (
           <RtcPlayer
+            username={username}
```

Comment:
IMO it's a very misleading name, as my first thought was that this is the username of the presenter. What do you think about changing it to `clientName` or even passing to the component flag whether the `isMe` or `isMyself` flag?